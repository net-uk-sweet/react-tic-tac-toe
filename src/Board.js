import React from 'react';

import { splitIntoChunks } from './helpers';
import Row from './Row';
import Square from './Square';

export default class Board extends React.Component {

    renderSquare(i, row, column) {
      const props = {
        highlight: this.props.winner && this.props.winner.includes(i),
        value: this.props.squares.getIn([i, 'token']),
        key: i + 1,
        onClick: () => this.props.onClick(i, row, column)
      }
      return <Square {...props} />;
    }
  
    renderRow(rowData, rowCount) {
      return (
        <Row key={rowCount}>
          { rowData.map((el, index) => this.renderSquare((rowCount * this.props.columns) + index, rowCount, index + 1)) }
        </Row>
      );
    }
  
    render() {
      const rows = splitIntoChunks(this.props.squares, this.props.columns).map((row, index) => {
        return this.renderRow(row, index);
      });
  
      return (
        <div>
          {rows}
        </div>
      )
    }
  }