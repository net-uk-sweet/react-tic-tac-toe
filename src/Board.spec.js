import Immutable from 'immutable';
import React from 'react';
import { mount, shallow } from 'enzyme';

import Board from './Board';

let shared;
let mockedClick;

beforeEach(() => {
    mockedClick = jest.fn();
    shared = {
        props: {
            winner: [0, 1, 2],
            squares: Immutable.fromJS(Array(9).fill(null)),
            columns: 3,
            onClick: mockedClick
        }
    }
    shared.wrapper = shallow(<Board {...shared.props} />);
});

it('matches snapshot', () => {
    expect(shared.wrapper.getElement()).toMatchSnapshot();
});

it('calls the onClick callback when a square is clicked with the correct index, row and column', () => {
    shared.wrapper = mount(<Board {...shared.props} />);
    shared.wrapper.find('button').at(0).simulate('click');
    expect(mockedClick).toHaveBeenCalledWith(0, 1, 1);
});
