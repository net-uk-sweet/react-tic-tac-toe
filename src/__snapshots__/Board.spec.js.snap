// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  Array [
    <Row>
      Immutable.List [
          <Square
              highlight={true}
              onClick={[Function]}
              value={undefined}
          />,
          <Square
              highlight={true}
              onClick={[Function]}
              value={undefined}
          />,
          <Square
              highlight={true}
              onClick={[Function]}
              value={undefined}
          />,
      ]
  </Row>,
    <Row>
      Immutable.List [
          <Square
              highlight={false}
              onClick={[Function]}
              value={undefined}
          />,
          <Square
              highlight={false}
              onClick={[Function]}
              value={undefined}
          />,
          <Square
              highlight={false}
              onClick={[Function]}
              value={undefined}
          />,
      ]
  </Row>,
    <Row>
      Immutable.List [
          <Square
              highlight={false}
              onClick={[Function]}
              value={undefined}
          />,
          <Square
              highlight={false}
              onClick={[Function]}
              value={undefined}
          />,
          <Square
              highlight={false}
              onClick={[Function]}
              value={undefined}
          />,
      ]
  </Row>,
  ]
</div>
`;
